src/sql/workbench/services/connection/browser/connectionController.ts

```diff
@@ -47,16 +47,16 @@ export class ConnectionController implements IConnectionComponentController {
 			onCreateNewServerGroup: () => this.onCreateNewServerGroup(),
 			onAdvancedProperties: () => this.handleOnAdvancedProperties(),
 			onSetAzureTimeOut: () => this.handleonSetAzureTimeOut(),
-			onFetchDatabases: (serverName: string, authenticationType: string, userName?: string, password?: string, azureAccount?: string) => this.onFetchDatabases(
-				serverName, authenticationType, userName, password, azureAccount).then(result => {
+			onFetchDatabases: (serverName: string, authenticationType: string, userName?: string, password?: string, azureAccount?: string, dstsToken?: string) => this.onFetchDatabases(
```

Comment:
And same kind of concern here - why would we treat dsts as special here instead of just having a generic concept of a security token that's added to requests to authenticate.

How is this token applied to the request anyways? Is there some property in the connection string that is set?

Reply:
There is a property that's set on the connection string, but the connection string is actually generated in the Kusto tool service. Rather than send a connection string, we actually send a cluster name, database name, and all the necessary information to generate a connection string.